Flaky-test: StreamingEntryReaderTests.testCanReadEntryFromMLedgerSizeExceededLimit

StreamingEntryReaderTests is flaky. The testCanReadEntryFromMLedgerSizeExceededLimit test method fails sporadically.

It seems that the 300 millisecond timeout for async operations is too short.

example failure

```
org.awaitility.core.ConditionTimeoutException: Condition with lambda expression in
 org.apache.pulsar.broker.service.streamingdispatch.StreamingEntryReaderTests that uses 
java.util.concurrent.atomic.AtomicBoolean was not fulfilled within 300 milliseconds.
 	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:165)
 	at org.awaitility.core.CallableCondition.await(CallableCondition.java:78)
 	at org.awaitility.core.CallableCondition.await(CallableCondition.java:26)
 	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:895)
 	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:864)
 	at org.apache.pulsar.broker.service.streamingdispatch.StreamingEntryReaderTests.testCanReadEntryFromMLedgerSizeExceededLimit(StreamingEntryReaderTests.java:182)
```

<details>
<summary>Full exception stacktrace</summary>
<code><pre>
 [INFO] Running org.apache.pulsar.broker.service.streamingdispatch.StreamingEntryReaderTests
 [ERROR] Tests run: 7, Failures: 1, Errors: 0, Skipped: 1, Time elapsed: 3.543 s <<< FAILURE! - in org.apache.pulsar.broker.service.streamingdispatch.StreamingEntryReaderTests
 [ERROR] testCanReadEntryFromMLedgerSizeExceededLimit(org.apache.pulsar.broker.service.streamingdispatch.StreamingEntryReaderTests)  Time elapsed: 0.346 s  <<< FAILURE!
 org.awaitility.core.ConditionTimeoutException: Condition with lambda expression in org.apache.pulsar.broker.service.streamingdispatch.StreamingEntryReaderTests that uses java.util.concurrent.atomic.AtomicBoolean was not fulfilled within 300 milliseconds.
 	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:165)
 	at org.awaitility.core.CallableCondition.await(CallableCondition.java:78)
 	at org.awaitility.core.CallableCondition.await(CallableCondition.java:26)
 	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:895)
 	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:864)
 	at org.apache.pulsar.broker.service.streamingdispatch.StreamingEntryReaderTests.testCanReadEntryFromMLedgerSizeExceededLimit(StreamingEntryReaderTests.java:182)
 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 	at java.lang.reflect.Method.invoke(Method.java:498)
 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
 	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
 	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
 	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
 	at java.lang.Thread.run(Thread.java:748)
</pre></code>
</details>

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/streamingdispatch/StreamingEntryReader.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/streamingdispatch/StreamingEntryReader.java
@@ -74,7 +74,7 @@ public class StreamingEntryReader implements AsyncCallbacks.ReadEntryCallback, W
             1, TimeUnit.SECONDS, 0, TimeUnit.MILLISECONDS);
 
     /**
-     * Read entries in streaming way, that said instead reading with micro batch and send entries to consumer after all
+     * Read entries in streaming way, that said instead of reading with micro batch and send entries to consumer after all
      * entries in the batch are read from ledger, this method will fire numEntriesToRead requests to managedLedger
      * and send entry to consumer whenever it is read && all entries before it have been sent to consumer.
      * @param numEntriesToRead number of entry to read from ledger.

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/streamingdispatch/StreamingEntryReader.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/streamingdispatch/StreamingEntryReader.java
@@ -74,8 +74,8 @@ public class StreamingEntryReader implements AsyncCallbacks.ReadEntryCallback, W
             1, TimeUnit.SECONDS, 0, TimeUnit.MILLISECONDS);
 
     /**
-     * Read entries in streaming way, that said instead of reading with micro batch and send entries to consumer after all
-     * entries in the batch are read from ledger, this method will fire numEntriesToRead requests to managedLedger
+     * Read entries in streaming way, that said instead of reading with micro batch and send entries to consumer after
+     * all entries in the batch are read from ledger, this method will fire numEntriesToRead requests to managedLedger
      * and send entry to consumer whenever it is read && all entries before it have been sent to consumer.
      * @param numEntriesToRead number of entry to read from ledger.
      * @param maxReadSizeByte maximum byte will be read from ledger.

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/service/streamingdispatch/StreamingEntryReaderTests.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/service/streamingdispatch/StreamingEntryReaderTests.java
@@ -179,7 +179,7 @@ public class StreamingEntryReaderTests extends MockedBookKeeperTestCase {
 
         // Only 2 entries should be read with this request.
         streamingEntryReader.asyncReadEntries(6, size * 2 + 1, null);
-        await().atMost(300, TimeUnit.MILLISECONDS).until(() -> readComplete.get());
+        await().atMost(1000, TimeUnit.MILLISECONDS).until(() -> readComplete.get());
         assertEquals(entries.size(), 2);
         // Assert cursor's read position has been properly updated to the third entry, since we should only read
         // 2 retries with previous request
@@ -187,13 +187,13 @@ public class StreamingEntryReaderTests extends MockedBookKeeperTestCase {
         reset(ledger);
         readComplete.set(false);
         streamingEntryReader.asyncReadEntries(6, size * 2 + 1, null);
-        await().atMost(300, TimeUnit.MILLISECONDS).until(() -> readComplete.get());
+        await().atMost(1000, TimeUnit.MILLISECONDS).until(() -> readComplete.get());
         readComplete.set(false);
         streamingEntryReader.asyncReadEntries(6, size * 2 + 1, null);
-        await().atMost(300, TimeUnit.MILLISECONDS).until(() -> readComplete.get());
+        await().atMost(1000, TimeUnit.MILLISECONDS).until(() -> readComplete.get());
         readComplete.set(false);
         streamingEntryReader.asyncReadEntries(6, size * 2 + 1, null);
-        await().atMost(300, TimeUnit.MILLISECONDS).until(() -> readComplete.get());
+        await().atMost(1000, TimeUnit.MILLISECONDS).until(() -> readComplete.get());
         assertEquals(cursor.getReadPosition(), ledger.getNextValidPosition((PositionImpl) positions.peek()));
         assertEquals(entries.size(), 8);
         for (int i = 0; i < entries.size(); i++) {
@@ -226,11 +226,11 @@ public class StreamingEntryReaderTests extends MockedBookKeeperTestCase {
         ).when(mockDispatcher).readEntryComplete(any(Entry.class), any(PendingReadEntryRequest.class));
 
         streamingEntryReader.asyncReadEntries(5,  100, null);
-        await().atMost(300, TimeUnit.MILLISECONDS).until(() -> entryCount.get() == 5);
+        await().atMost(500, TimeUnit.MILLISECONDS).until(() -> entryCount.get() == 5);
         assertEquals(cursor.getReadPosition(), ledger.getNextValidPosition((PositionImpl) positions.peek()));
         streamingEntryReader.asyncReadEntries(5, 100, null);
         // We only write 7 entries initially so only 7 entries can be read.
-        await().atMost(300, TimeUnit.MILLISECONDS).until(() -> entryCount.get() == 7);
+        await().atMost(500, TimeUnit.MILLISECONDS).until(() -> entryCount.get() == 7);
         // Add new entry and await for it to be send to reader.
         entryProcessed.set(false);
         ledger.addEntry("message-7".getBytes(Encoding));

